src/Files.App/Filesystem/FilesystemOperations/FilesystemOperations.cs

```diff
@@ -1,996 +1,776 @@
+            var collisionsNoSkip = collisions.Where(c => c != FileNameConflictResolveOptionType.Skip);
 
-                            break;
-                        }
+            var operationID = Guid.NewGuid().ToString();
+            using var r = cancellationToken.Register(CancelOperation, operationID, false);
 
-                    case FilesystemItemType.Directory:
-                        {
-                            var fsFolderResult = await associatedInstance.FilesystemViewModel.GetFolderFromPathAsync(PathNormalization.GetParentDir(source.Path));
-                            fsResult = fsFolderResult;
-                            if (fsResult)
-                            {
-                                var fsCreateResult = await FilesystemTasks.Wrap(() => fsFolderResult.Result.CreateFolderAsync(Path.GetFileName(source.Path), CreationCollisionOption.GenerateUniqueName).AsTask());
-                                fsResult = fsCreateResult;
-                                item = fsCreateResult.Result.FromStorageItem();
-                            }
-                            if (fsResult == FileSystemStatusCode.Unauthorized)
-                            {
-                                // Can't do anything, already tried with admin FTP
-                            }
-                            break;
-                        }
+            EventHandler<Dictionary<string, JsonElement>> handler = (s, e) => OnProgressUpdated(s, e, operationID, progress);
 
-                    default:
-                        Debugger.Break();
-                        break;
-                }
+            var sourceReplace = sourceNoSkip.Zip(collisionsNoSkip, (src, coll) => new { src, coll }).Where(item => item.coll == FileNameConflictResolveOptionType.ReplaceExisting).Select(item => item.src);
+            var destinationReplace = destinationNoSkip.Zip(collisionsNoSkip, (src, coll) => new { src, coll }).Where(item => item.coll == FileNameConflictResolveOptionType.ReplaceExisting).Select(item => item.src);
+            var sourceRename = sourceNoSkip.Zip(collisionsNoSkip, (src, coll) => new { src, coll }).Where(item => item.coll != FileNameConflictResolveOptionType.ReplaceExisting).Select(item => item.src);
+            var destinationRename = destinationNoSkip.Zip(collisionsNoSkip, (src, coll) => new { src, coll }).Where(item => item.coll != FileNameConflictResolveOptionType.ReplaceExisting).Select(item => item.src);
 
-                errorCode?.Report(fsResult);
-                if (item != null)
+            var result = (FilesystemResult)true;
+            var copyResult = new ShellOperationResult();
+            if (sourceRename.Any())
+            {
+                var (status, response) = await connection.SendMessageForResponseAsync(new ValueSet()
```

Comment:
Several of those constructs are repeated in the code. Maybe a method to regroup their creation into one method taking arguments to customize the message in Entry could be worth the trouble to set up?